ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/text_embedding/TextEmbeddingModel.java

```diff
@@ -0,0 +1,241 @@
+                    if (pathFile.exists()) {
+                        FileUtils.deleteDirectory(pathFile);
+                    }
+                    ZipUtils.unzip(new FileInputStream(modelZipFile), modelPath);
+                    boolean findModelFile = false;
+                    for (File file : pathFile.listFiles()) {
+                        String name = file.getName();
+                        if (name.endsWith(PT) || name.endsWith(ONNX)) {
+                            if (findModelFile) {
+                                throw new IllegalArgumentException("found multiple models");
+                            }
+                            findModelFile = true;
+                            String suffix = name.substring(name.lastIndexOf("."));
+                            file.renameTo(new File(modelPath.resolve(modelName + suffix).toUri()));
+                        }
+                    }
+                    Map<String, Object> arguments = new HashMap<>();
+                    Criteria.Builder<Input, Output> criteriaBuilder = Criteria.builder()
+                            .setTypes(Input.class, Output.class)
+                            .optApplication(Application.UNDEFINED)
+                            .optArguments(arguments)
+                            .optEngine(engine)
+                            .optModelPath(modelPath);
+                    TextEmbeddingModelConfig textEmbeddingModelConfig = (TextEmbeddingModelConfig) modelConfig;
+                    TextEmbeddingModelConfig.FrameworkType transformersType = textEmbeddingModelConfig.getFrameworkType();
+                    if (ONNX_ENGINE.equals(engine)) { //ONNX
+                        criteriaBuilder.optTranslator(new ONNXSentenceTransformerTextEmbeddingTranslator());
+                    } else { // pytorch
+                        if (transformersType == SENTENCE_TRANSFORMERS) {
+                            criteriaBuilder.optTranslator(new SentenceTransformerTextEmbeddingTranslator());
+                        } else {
+                            criteriaBuilder.optTranslatorFactory(new HuggingfaceTextEmbeddingTranslatorFactory());
+                        }
+                    }
+                    Criteria<Input, Output> criteria = criteriaBuilder.build();
+                    ZooModel<Input, Output> model = criteria.loadModel();
+                    Predictor<Input, Output> predictor = model.newPredictor();
+                    this.predictor = predictor;
+                    this.model = model;
+                    return null;
+                } catch (Exception e) {
+                    String errorMessage = "Failed to load model " + modelName + ", version: " + version;
+                    log.error(errorMessage, e);
+                    close();
+                    throw new MLException(errorMessage, e);
+                } finally {
+                    deleteFileQuietly(getLoadModelPath(modelId));
```

Comment:
So are we deleting the model file even if we load the model successfully?

Reply:
Yes, the load model path is just temporary folder.